pkg/storage/etcd/rbac_store.go

```diff
@@ -104,6 +220,7 @@ func (e *EtcdStore) GetRoleBinding(ctx context.Context, ref *corev1.Reference) (
 	if err := storage.ApplyRoleBindingTaints(ctx, e, roleBinding); err != nil {
```

Comment:
Question about role binding taints - why does the etcd and jetstream stores apply the taints in `GetRoleBinding` and not the crd store? 

My update code calls `GetRoleBinding` to perform an update with etcd or jetstream, and I end up with duplicate taints caused by `ApplyRoleBindingTaints` `rb.Taints = append(rb.Taints, "role not found")`.
![image](https://github.com/rancher/opni/assets/84587295/4e0cafa5-d008-4d9d-b913-6c97ab03e274)

 The crd rbac_store test doesn't even output taints if I get the role binding in a test.
```
Running Suite: CRDs Storage Suite - /home/ubuntu/opni/pkg/storage/crds
• [FAILED] [0.025 seconds]
RBAC Store Role Bindings [It] should handle taints for multiple role binding updates [integration, slow]
/home/ubuntu/opni/pkg/test/conformance/storage/rbac_store.go:219

  [FAILED] Expected
      <[]string | len:0, cap:0>: nil
  to equal
      <[]string | len:2, cap:2>: ["role not found", "no subjects"]
```
What's the expected behaviour?

Reply:
Role binding taints are purely an informative feature, the taints are computed on-demand and injected into the message before it's sent to the caller.